File: Practice/atcoder/ABC/146/src/c.py
# 超特殊なにぶたんを作る

a, b, x = map(int, input().split())


def keta(x):
    c = 0
    while x > 0:
        x //= 10
        c += 1
    return c


def super_nibutan(a, b, x):
    l, r = 0, x
    while l < r:
        # print("l, r = {}, {}".format(l, r))
        mid = (l + r) // 2
        if a * mid + b * keta(mid) > x:
            r = mid
        else:
            l = mid + 1
    l -= 1
    # print(
    #     "a*n + b*d(n) = {} * {} + {} * {} = {}".format(
    #         a, l, b, keta(l), a * l + b * keta(l)
    #     )
    # )
    return min(l, 10 ** 9)


print(super_nibutan(a, b, x))

File: Practice/atcoder/ABC/146/src/c.go
// にぶたんの臭いがする
// 買える最大値の誤読で数十分ほど椅子を温めた
// 何度もWAをして結局写経した
// コード中の①~③が大事っぽい？？？？

package main

import "fmt"

var a, b, x int

func main() {
	fmt.Scan(&a, &b, &x)
	fmt.Println(nibutan(x))
}

func keta(k int) int {
	ret := 0
	for k > 0 {
		ret += 1
		k /= 10
	}
	return ret
}

func nibutan(p int) int {
	l, r := 0, 1000000001 // ① rは上限+1
	// for r-l > 1 {
	for r-l > 1 { // ② 比較は差が1より上
		mid := (l + r) / 2
		f := a*mid + b*keta(mid)
		if p < f {
			// でかすぎる
			r = mid
		} else {
			// まだいける
			l = mid // ③ +1はしない
		}
	}
	return l
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}
